ec2/main.tf

```diff
@@ -0,0 +1,136 @@
+#Create Security Groups for public instances
+resource "aws_security_group" "public_ec2_sg" {
+  name = "public_ec2_sg"
+  description = "Allows TLS inbound traffic and outbound traffic"
+  vpc_id = aws_vpc.practice_vpc.id
+
+/*
+By default, AWS creates an ALLOW ALL egress rule when creating 
+a new Security Group inside of a VPC. When creating a new 
+Security Group inside a VPC, Terraform will remove this default 
+rule, and require you specifically re-create it if you desire 
+that rule. We feel this leads to fewer surprises in terms of 
+controlling your egress rules. If you desire this rule to be in 
+place, you can use this egress block:
+*/
+
+  egress {
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  tags = {
+    Name = "public_ec2_security_group"
+    vpc = "practice_vpc"
```

Comment:
Remove the vpc tag , i don't think it's useful.

Reply:
across board?

Reply:
yes